src/sub.c

```diff
@@ -1124,6 +1124,54 @@ natsSubscription_QueuedMsgs(natsSubscription *sub, uint64_t *queuedMsgs)
     return s;
 }
 
+int64_t
+natsSubscription_GetID(natsSubscription* sub)
+{
+    if (sub == NULL)
+        return 0;
+
+    natsSub_Lock(sub);
+
+    if (sub->closed)
+    {
+        natsSub_Unlock(sub);
+        return 0;
+    }
+
+    int64_t id = 0;
+    SUB_DLV_WORKER_LOCK(sub);
+    id = sub->sid;
+    SUB_DLV_WORKER_UNLOCK(sub);
+
+    natsSub_Unlock(sub);
+
+    return id;
+}
+
+const char*
+natsSubscription_GetSubject(natsSubscription* sub)
+{
+    if (sub == NULL)
+        return NULL;
+
+    natsSub_Lock(sub);
+
+    if (sub->closed)
+    {
+        natsSub_Unlock(sub);
+        return NULL;
+    }
+
+    const char* subject = NULL;
```

Comment:
Same, please declare at top, and remove SUB_DLV_WORKER_LOCK/SUB_DLV_WORKER_UNLOCK